src/services/expenses.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+const { createId } = require('../utils/createId');
+
+let expenses = [];
+
+const getInitial = () => {
+  expenses = [];
+};
+
+const getExpenses = (userId, categories, from, to) => {
+  let filteredExpenses = expenses;
+
+  if (!isNaN(+userId)) {
+    filteredExpenses = filteredExpenses
+      .filter(expense => expense.userId === +userId);
+  }
+
+  if (categories && categories.length) {
+    filteredExpenses = filteredExpenses
+      .filter(expense => categories.includes(expense.category));
+  }
+
+  if (from) {
+    filteredExpenses = filteredExpenses
+      .filter(expense => Date.parse(expense.spentAt) >= Date.parse(from));
+  }
+
+  if (to) {
+    filteredExpenses = filteredExpenses
+      .filter(expense => Date.parse(expense.spentAt) <= Date.parse(to));
```

Comment:
so if you have all params, you will iterate 4 times?
better to do this logic due to 1 iteration only